File: frontend/src/objects/merchant/store/StoreIdentity.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  CuisineLabel,
  DisplayText,
  PersonName,
  StoreId,
} from '@/objects/core/SharedObjects'

export type StoreIdentity = {
  id: StoreId
  merchantName: PersonName
  name: DisplayText
  category: DisplayText
  cuisine: CuisineLabel
}

File: backend/src/main/scala/services/merchant/objects/StoreIdentity.scala
package services.merchant.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class StoreIdentity(
    id: StoreId,
    merchantName: PersonName,
    name: DisplayText,
    category: DisplayText,
    cuisine: CuisineLabel,
)
object StoreIdentity:
  given Encoder[StoreIdentity] = deriveEncoder
  given Decoder[StoreIdentity] = deriveDecoder
